test/data_sync_config_test.cpp

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
Just suggestion: It would be great if you added negative test cases by providing incorrect or unsupported values for sync type, direction, and duration interval.

Reply:
Done.